hivemq-edge/src/main/java/com/hivemq/api/auth/provider/impl/ldap/LdapConnectionProperties.java

```diff
@@ -0,0 +1,341 @@
+     * @param tlsMode               The TLS/SSL mode to use
+     * @param trustStorePath        The path to the truststore file
+     * @param trustStorePassword    The password for the truststore
+     * @param trustStoreType        The type of the truststore
+     * @param connectTimeoutMillis  Connection timeout in milliseconds (0 = use default)
+     * @param responseTimeoutMillis Response timeout in milliseconds (0 = use default)
+     * @param userDnTemplate        The DN template for resolving usernames
+     * @param baseDn                The base DN of the LDAP directory
+     */
+    public LdapConnectionProperties(
+            final @NotNull String host,
+            final int port,
+            final @NotNull TlsMode tlsMode,
+            final @Nullable String trustStorePath,
+            final @Nullable String trustStorePassword,
+            final @Nullable String trustStoreType,
+            final int connectTimeoutMillis,
+            final int responseTimeoutMillis,
+            final @NotNull String userDnTemplate,
+            final @NotNull String baseDn) {
+        this(host, port, tlsMode, trustStorePath, trustStorePassword, trustStoreType, connectTimeoutMillis, responseTimeoutMillis, userDnTemplate, baseDn, false);
+    }
+
+    /**
+     * Creates LdapConnectionProperties from XML configuration entity.
+     * <p>
+     * This factory method converts an {@link com.hivemq.configuration.entity.api.LdapAuthenticationEntity}
+     * (loaded from config.xml) into runtime connection properties.
+     *
+     * @param entity The LDAP authentication configuration from XML
+     * @return Configured LdapConnectionProperties for runtime use
+     * @throws IllegalArgumentException if the entity configuration is invalid
+     */
+    public static @NotNull LdapConnectionProperties fromEntity(
+            final @NotNull LdapAuthenticationEntity entity) {
+        // Parse TLS mode from string
+        final TlsMode tlsMode;
+        try {
+            tlsMode = TlsMode.valueOf(entity.getTlsMode().toUpperCase());
+        } catch (final IllegalArgumentException e) {
+            throw new IllegalArgumentException(
+                    "Invalid TLS mode: " + entity.getTlsMode() + ". Must be one of: NONE, LDAPS, START_TLS", e);
+        }
+
+        // Determine port: use configured port, or default based on TLS mode if 0
+        final int port = entity.getPort() > 0 ? entity.getPort() : tlsMode.getDefaultPort();
+
+        // Extract TLS configuration if present
+        final String trustStorePath;
+        final String trustStorePassword;
+        final String trustStoreType;
+        if (entity.getTls() != null) {
+            trustStorePath = entity.getTls().getTrustStorePath();
+            trustStorePassword = entity.getTls().getTrustStorePassword();
+            trustStoreType = entity.getTls().getTrustStoreType();
+        } else {
+            trustStorePath = null;
+            trustStorePassword = null;
```

Comment:
yeah, passwords stored as Strings, perhaps we need to create a new ticket to go after these through the codebase.

Reply:
Yep, agreed. I just checked and all passwords coming from the xml config are first put into strings.
We need some dedicated handling for that.

Reply:
Ticket added: https://hivemq.kanbanize.com/ctrl_board/57/cards/37505/details/